sql/planbuilder/factory.go

```diff
@@ -175,3 +177,29 @@ func (f *factory) buildTableAlias(name string, child sql.Node) (plan.TableIdNode
 		}
 	}
 }
+
+func (f *factory) buildDistinct(child sql.Node) sql.Node {
+	if proj, isProj := child.(*plan.Project); isProj {
```

Comment:
the one thing i would add is a quick `// from => to` docstring to make the tranform clearer